<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowClass</class>
 <widget class="QMainWindow" name="MainWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>658</width>
    <height>523</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>658</width>
    <height>523</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="minimumSize">
    <size>
     <width>658</width>
     <height>444</height>
    </size>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QHBoxLayout" name="renderingLayout">
      <item>
       <widget class="RenderingWidget" name="openGLWidget">
        <property name="enabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>658</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="openAction"/>
   <addaction name="saveAction"/>
   <addaction name="mntnAction"/>
   <addaction name="separator"/>
   <addaction name="lightAction"/>
   <addaction name="viewLineAction"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="openAction">
   <property name="icon">
    <iconset resource="mainwindow.qrc">
     <normaloff>:/images/file-open</normaloff>:/images/file-open</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="toolTip">
    <string>Open Model File</string>
   </property>
  </action>
  <action name="saveAction">
   <property name="icon">
    <iconset resource="mainwindow.qrc">
     <normaloff>:/images/file-save</normaloff>:/images/file-save</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save Model File</string>
   </property>
  </action>
  <action name="mntnAction">
   <property name="icon">
    <iconset resource="mainwindow.qrc">
     <normaloff>:/images/file-maintenance</normaloff>:/images/file-maintenance</iconset>
   </property>
   <property name="text">
    <string>Maintenance</string>
   </property>
   <property name="toolTip">
    <string>Maintenance Model File</string>
   </property>
  </action>
  <action name="lightAction">
   <property name="icon">
    <iconset resource="mainwindow.qrc">
     <normaloff>:/images/model-light</normaloff>:/images/model-light</iconset>
   </property>
   <property name="text">
    <string>Light</string>
   </property>
   <property name="toolTip">
    <string>set Light</string>
   </property>
  </action>
  <action name="viewLineAction">
   <property name="icon">
    <iconset resource="mainwindow.qrc">
     <normaloff>:/images/model-line</normaloff>:/images/model-line</iconset>
   </property>
   <property name="text">
    <string>Line View</string>
   </property>
   <property name="toolTip">
    <string>set Line in Model</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>RenderingWidget</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">renderingwidget.h</header>
   <slots>
    <slot>cutinPieces()</slot>
    <slot>ReadMesh()</slot>
    <slot>renderdoHatch()</slot>
    <slot>CheckGrid()</slot>
    <slot>AutoSupport()</slot>
    <slot>DeleteSupport()</slot>
    <slot>CheckDrawEdge()</slot>
    <slot>CheckDrawFace()</slot>
    <slot>AddLineSupport()</slot>
    <slot>AddPointSupport()</slot>
    <slot>SetBackground()</slot>
    <slot>SelectFace()</slot>
    <slot>SetDirection()</slot>
    <slot>Export()</slot>
    <slot>WriteMesh()</slot>
    <slot>setLaserPower(int)</slot>
    <slot>setLaserSpeed(int)</slot>
    <slot>SetSliceCheckId(int)</slot>
    <slot>CheckLight()</slot>
    <slot>setHatchType(int)</slot>
    <slot>setOffset(double)</slot>
    <slot>setfieldWidth(double)</slot>
    <slot>setfieldHeight(double)</slot>
    <slot>setlineOverlap(int)</slot>
    <slot>setfieldOverlap(double)</slot>
    <slot>setThickness(double)</slot>
    <slot>CheckSetFace()</slot>
    <slot>setLaserPowerPolygon(int)</slot>
    <slot>setLaserSpeedPolygon(int)</slot>
    <slot>setAngle(int)</slot>
    <slot>setThreshold(double)</slot>
    <slot>setGap(double)</slot>
    <slot>setSeglength(double)</slot>
    <slot>setReso(double)</slot>
    <slot>setVerticalgap(double)</slot>
    <slot>CheckRegion(bool)</slot>
    <slot>CheckSupport(bool)</slot>
    <slot>setFildID(int)</slot>
    <slot>setPointD(double)</slot>
    <slot>setPointH(double)</slot>
    <slot>setLineD(double)</slot>
    <slot>setLineH(double)</slot>
    <slot>FindRegion()</slot>
    <slot>CheckRotateModel(bool)</slot>
    <slot>SetScaleT(double)</slot>
    <slot>SclaleModel()</slot>
    <slot>SetAllHatch(bool)</slot>
    <slot>SetSliceCheckIdText(int)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="mainwindow.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>openAction</sender>
   <signal>triggered()</signal>
   <receiver>openGLWidget</receiver>
   <slot>ReadMesh()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>328</x>
     <y>278</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveAction</sender>
   <signal>triggered()</signal>
   <receiver>openGLWidget</receiver>
   <slot>WriteMesh()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>328</x>
     <y>278</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lightAction</sender>
   <signal>triggered()</signal>
   <receiver>openGLWidget</receiver>
   <slot>CheckLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>328</x>
     <y>278</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>viewLineAction</sender>
   <signal>triggered(bool)</signal>
   <receiver>openGLWidget</receiver>
   <slot>CheckDrawEdge()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>328</x>
     <y>278</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
